src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeed.cs

```diff
@@ -190,38 +190,52 @@ public string GeneratePackageServiceIndex(string redirectUrl, IEnumerable<string
         private string GenerateRootServiceIndex(string redirectUrl)
         {
             string pathToTempFolder = Path.Combine(IndexDirectory, redirectUrl);
-            if (Directory.Exists(pathToTempFolder))
-            {
-                Directory.Delete(pathToTempFolder, true);
-            }
-            Directory.CreateDirectory(pathToTempFolder);
             string rootIndexJsonLocation = Path.Combine(pathToTempFolder, "index.json");
-            using (FileStream fs = File.OpenWrite(rootIndexJsonLocation))
-            using (var streamWriter = new StreamWriter(fs))
-            using (JsonTextWriter writer = new JsonTextWriter(streamWriter))
-            {
-                writer.WriteStartObject();
-                writer.WritePropertyName("version");
-                writer.WriteRawValue("\"3.0.0\"");
-                writer.WritePropertyName("resources");
-                writer.WriteStartArray();
-                writer.WriteStartObject();
-                writer.WritePropertyName("@id");
-                writer.WriteRawValue($"\"{FeedUrl}\"");
-                writer.WritePropertyName("@type");
-                writer.WriteRawValue("\"PackageBaseAddress/3.0.0\"");
-                writer.WritePropertyName("comment");
-                writer.WriteRawValue("\"Base URL of Azure storage where NuGet package registration info for intermediaries is stored.\"");
-                writer.WriteEndObject();
-                writer.WriteEndArray();
-                writer.WritePropertyName("@context");
-                writer.WriteStartObject();
-                writer.WritePropertyName("@vocab");
-                writer.WriteRawValue("\"http://schema.nuget.org/schema#\"");
-                writer.WritePropertyName("comment");
-                writer.WriteRawValue("\"http://www.w3.org/2000/01/rdf-schema#comment\"");
-                writer.WriteEndObject();
-                writer.WriteEndObject();
+            int retry = 5;
+            bool isFileCreated = false;
+            while (!isFileCreated && --retry > 0) { 
```

Comment:
delays are generally helpful in retry loops to help avoid transient issues.

Reply:
Added in the catch